numba/mviewbuf.c

```diff
@@ -1,5 +1,199 @@
 #include "_pymodule.h"
 
+/**
+ * A BufferProxy wraps a read-only buffer in a writable buffer. This is required
+ * for NumPy void and record types, which only expose a read-only buffer. Note
+ * that wrapping a read-only buffer may have some surprising effects on buffers
+ * which expect the data from their read-only buffer not to be modified.
+ */
+typedef struct {
+    PyObject_HEAD
+    PyObject *wrapped;
+    int ob_exports;
+    Py_buffer wrapped_buf;
+#if (PY_MAJOR_VERSION < 3)
+    void *wrapped_ptr;
+    Py_ssize_t wrapped_len;
+    int ob_old_exports;
+#endif
+} BufferProxyObject;
+
+static int
+BufferProxyObject_getbuffer(BufferProxyObject *self, Py_buffer *view,
+                            int flags) {
+    int supported_flags = PyBUF_WRITABLE|PyBUF_ND|PyBUF_STRIDES|PyBUF_FORMAT;
+    int wrapped_flags = supported_flags & ~PyBUF_WRITABLE;
+
+    if (flags & ~supported_flags) {
+        PyErr_SetString(PyExc_BufferError, "Unsupported flags for BufferProxy");
+        view->obj = NULL;
```

Comment:
Looks like "view->obj = NULL" could be factored out at the beginning of the function :-)